buildSrc/src/main/java/org/opensearch/gradle/internal/BwcSetupExtension.java

```diff
@@ -80,77 +81,83 @@ public BwcSetupExtension(
-            // TODO revisit
-            loggedExec.dependsOn("checkoutBwcBranch");
-            loggedExec.setSpoolOutput(true);
-            loggedExec.setWorkingDir(checkoutDir.get());
-            loggedExec.doFirst(t -> {
-                // Execution time so that the checkouts are available
-                String javaVersionsString = readFromFile(new File(checkoutDir.get(), ".ci/java-versions.properties"));
-                loggedExec.environment(
-                    "JAVA_HOME",
-                    getJavaHome(
-                        Integer.parseInt(
-                            Arrays.asList(javaVersionsString.split("\n"))
-                                .stream()
-                                .filter(l -> l.trim().startsWith("OPENSEARCH_BUILD_JAVA="))
-                                .map(l -> l.replace("OPENSEARCH_BUILD_JAVA=java", "").trim())
-                                .map(l -> l.replace("OPENSEARCH_BUILD_JAVA=openjdk", "").trim())
-                                .collect(Collectors.joining("!!"))
-                        )
-                    )
-                );
-                loggedExec.environment(
-                    "RUNTIME_JAVA_HOME",
-                    getJavaHome(
-                        Integer.parseInt(
-                            Arrays.asList(javaVersionsString.split("\n"))
-                                .stream()
-                                .filter(l -> l.trim().startsWith("OPENSEARCH_RUNTIME_JAVA="))
-                                .map(l -> l.replace("OPENSEARCH_RUNTIME_JAVA=java", "").trim())
-                                .map(l -> l.replace("OPENSEARCH_RUNTIME_JAVA=openjdk", "").trim())
-                                .collect(Collectors.joining("!!"))
-                        )
-                    )
-                );
-            });
-
-            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                loggedExec.executable("cmd");
-                loggedExec.args("/C", "call", new File(checkoutDir.get(), "gradlew").toString());
-            } else {
-                loggedExec.executable(new File(checkoutDir.get(), "gradlew").toString());
-            }
-            if (project.getGradle().getStartParameter().isOffline()) {
-                loggedExec.args("--offline");
-            }
-            // TODO resolve
-            String buildCacheUrl = System.getProperty("org.opensearch.build.cache.url");
-            if (buildCacheUrl != null) {
-                loggedExec.args("-Dorg.opensearch.build.cache.url=" + buildCacheUrl);
-            }
+        return project.getTasks().register(name, LoggedExec.class, new Action<LoggedExec>() {
+            @Override
+            public void execute(LoggedExec loggedExec) {
+                // TODO revisit
+                loggedExec.dependsOn("checkoutBwcBranch");
+                loggedExec.setSpoolOutput(true);
+                loggedExec.setWorkingDir(checkoutDir.get());
+                loggedExec.doFirst(new Action<Task>() {
```

Comment:
Reason: Using Java lambdas is not supported as task inputs. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implementation_unknown for more details about this problem.